Lib/test/test_socket.py

```diff
@@ -2356,6 +2356,41 @@ def testRecvFromNegative(self):
     def _testRecvFromNegative(self):
         self.cli.sendto(MSG, 0, (HOST, self.port))
 
+
+@requireSocket("AF_BLUETOOTH", "SOCK_STREAM", "BTPROTO_RFCOMM")
+class BasicBluetoothTest(unittest.TestCase):
+    def setUp(self):
+        self.bs  = socket.socket(socket.AF_BLUETOOTH,
+                                 socket.SOCK_STREAM,
+                                 socket.BTPROTO_RFCOMM)
+
+    def tearDown(self):
+        self.bs.close()
+
+    def test_address_invalid(self):
+        addr = 'ff:ff:ff:ff:fffa'
+        msg = '"{addr} is not a valid address.'.format(addr=addr)
+        with self.assertRaisesRegex(OSError, 'bad bluetooth address', msg=msg) as cm:
+            self.bs.connect((addr, 1))
+        self.assertIsNone(cm.exception.errno, msg)
```

Comment:
Shouldn't we check that that `errno` is some expected value? `EINVAL`? Also, why are you passing in `msg` here?